packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -408,13 +415,20 @@ export default class Extension {
         .createType('ExtrinsicPayload', payload, { version: payload.version })
         .sign(pair);
 
-    if (savePass) {
+    if (savePass && password) {
       // unlike queued.account.address the following
       // address is encoded with the default prefix
       // which what is used for password caching mapping
-      this.#cachedUnlocks[pair.address] = Date.now() + PASSWORD_EXPIRY_MS;
+
+      await chromeStorage.setItem('password-cache', (savedPasswords) => ({ ...savedPasswords, [pair.address]: { password, expiresAt: Date.now() + PASSWORD_EXPIRY_MS } }));
     } else {
-      pair.lock();
+      const savedPasswords = await chromeStorage.getItem('password-cache');
+
+      const remainingTime = (savedPasswords && savedPasswords[pair.address]?.expiresAt - Date.now()) || 0;
+
+      if (remainingTime <= 0) {
+        pair.lock();
+      }
```

Comment:
I think here we should:
* lock the pair
* forget the password

For me it doesn't matter if password expired or not, when user unchecks the save password checkbox we should clear pair / storage. And mind you he need to explicitly uncheck it, because it's checked by default as long as password is remebered

Reply:
The checkbox stats - Extended the period without password for 15 minutes, so for me it more make sense that:

1. checked = resets timer to 15 minuts
2. unchecked = keeps as it is.

If you think its better to make that if checkbox unchecked we clear pair / storage, then we should rename label of checkbox to something like: Keep password saved for 15 minutes.

or wdyt?

Reply:
I see, I haven't read the checkbox label :see_no_evil:
Alright, so
* in case `remainingTime <= 0` forget the password is missing
* I feel like we do all of this already in `refreshAccountPasswordCache`, don't we?

Jeez, this function is hella hard to read, I think I constantly catch myself not understanding what's going on (it was like this before, it's just changing it is minefield :sweat_smile: ).

Btw, I was wondering about it - if someone passes a wrong password to this function we are still going to use the cache one and approve the transaction. Though it should not happen... perhaps we should just always take `inputPassword` if set? And always validate it? Maybe that's unnecessary, just a thought

Reply:
I suggest to only keep one label: Save password for 15 minutes, and remove this extend password  wording.

I see scenario when person knows its gonna be his last transaction, and he wants to uncheck checkbox to make sure that his password is not saved anymore.

Reply:
Does this look right? 

```
  private async refreshAccountPasswordCache (pair: KeyringPair): Promise<{remainingTime: number, cachedPassword?: string}> {
    const { address } = pair;

    const savedPasswords = await chromeStorage.getItem('password-cache') ?? {};
    const expiresAt = savedPasswords?.[address]?.expiresAt ?? 0;
    const remainingTime = expiresAt - Date.now();
    const cachedPassword = savedPasswords?.[address]?.password;

    if (remainingTime > 0) {
      return { remainingTime, cachedPassword };
    }

    delete savedPasswords[address];
    await chromeStorage.setItem('password-cache', () => (savedPasswords));
    pair.lock();

   
``` return { remainingTime: 0 };
  }

```
    const { cachedPassword } = await this.refreshAccountPasswordCache(pair);

    const password = cachedPassword ?? inputPassword;
```